### PR TITLE
Don't upload to PyPI on workflow dispatch

### PR DESCRIPTION
**Pull request by astrofrog**
Just a small tweak to the publish workflow - now that when we do workflow_dispatch we default to building against Numpy dev, I don't think we want to upload to PyPI also when using workflow_dispatch. I think it makes sense to keep workflow_dispatch as a developer thing anyway, and not to make actual stable releases.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
That certainly makes sense to me, but I'll let @saimn review too, as perhaps I'm missing something.

**Review by pllim — CHANGES_REQUESTED** [review]
I am not sure if I like this now that I think about it more. In several occasions, I need to fix downstream job fast (or requested by another downstream maintainer), so I used the dispatch to generate new dev wheel on the fly.

**Comment by pllim** [comment]
Also I don't think this is upload to PyPI but to the Anaconda wheel repo, right?

**Review by pllim — APPROVED** [review]
No nvm... I missed that Anaconda is a different setting. Let's just get this in then. Thanks!

**Pull request merged**
